src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -5316,3 +5316,31 @@ void C2_MacroAssembler::udivmodL(Register rax, Register divisor, Register rdx, R
   bind(done);
 }
 #endif
+
+void C2_MacroAssembler::float_class_check_vfp(int opcode, Register dst, XMMRegister src, KRegister tmp) {
+  uint8_t imm8;
+  switch (opcode) {
+  // allows future usecases (ex: isNaN, isFinite)
+    case Op_IsInfiniteF:
+      imm8 = 0x18;
+      break;
+    default:
+      assert(false, "%s", NodeClassNames[opcode]);
+  }
+  vfpclassss(tmp, src, imm8);
+  kmovbl(dst, tmp);
+}
+
+void C2_MacroAssembler::double_class_check_vfp(int opcode, Register dst, XMMRegister src, KRegister tmp) {
+  uint8_t imm8;
```

Comment:
May be ok to move it back to instruction encoding block , only two instructions.

Reply:
That's true. The two instructions can be put in the instruction encoding block. Will do that.

Reply:
Also, in future if we want to add the support for isFinite() and IsNaN(), wouldn't it be better to have seperate macros?

Reply:
Updated the code with the suggested changes. Thanks Jatin and Vladimir!